src/libs/scene_utils/samples/cookie_factory_sample/sample.ts

```diff
@@ -0,0 +1,198 @@
+        .withPosition({ x: 0, y: 2.84, z: 0 });
+
+      // Prepare 3D Model
+      const mixerAssetFile = `${assetPath}CookieFactoryMixer.glb`;
+      const modelRefNode: ModelRefNode = assetToModelRef(mixerAssetFile, entitySummary.entityName);
+      modelRefNode.uploadModelFromLocalIfNotExist(mixerAssetFile);
+
+      // Add Tag to 3D Model node
+      modelRefNode.addChildNode(tagNode);
+
+      // Set Mixer Transform
+      processMixerTransform(modelRefNode, entitySummary.entityName, csvResult);
+
+      // Add 3D Model to the Scene
+      mixersNode.addChildNode(modelRefNode);
+    } else if (entitySummary.entityId.includes('WaterTank')) {
+      // Prepare Tag
+      const tagNode: TagNode = new TagNode('Tag');
+      const tagDataBinding = new DataBinding();
+
+      tagDataBinding
+        .withTargetEntityId(entitySummary.entityId)
+        .withTargetComponentName('AlarmComponent')
+        .withTargetProperty('alarm_status');
+
+      // Update the Tag Node with a position and the above data binding and rule
+      tagNode
+        .withDataBinding(tagDataBinding)
+        .withTarget(Target.INFO)
+        .withRuleId(alarmRuleName)
+        .withPosition({ x: 0, y: 0.86, z: 0 });
+
+      // Prepare 3D Model
+      const waterTankAssetFile = `${assetPath}CookieFactoryWaterTank.glb`;
+      const waterTankNode: ModelRefNode = assetToModelRef(waterTankAssetFile, 'WaterTank');
+      waterTankNode.uploadModelFromLocalIfNotExist(waterTankAssetFile);
+
+      waterTankNode.withPosition({ x: 32.6, y: 0, z: 47 });
+      waterTankNode.addChildNode(tagNode);
+
+      // Create an instance of Motion Indicator node
+      const motionIndicator: MotionIndicatorNode = new MotionIndicatorNode('MotionIndicator');
+
+      // Set Motion Indicator features and transform
+      motionIndicator
+        .setMotionIndicatorShape('LinearPlane')
+        .setMotionDefaultSpeed(1)
+        .setDefaultForegroundColor(0xffff00) // YELLOW
+        .setNumOfRepeatInY(1)
+        .withPosition({ x: 1.4, y: 3, z: -3 })
+        .withRotation({ x: 0, y: 270, z: 0 })
+        .withScale({ x: 7, y: 1, z: 1 });
+
+      waterTankNode.addChildNode(motionIndicator);
+
+      // Create a Model Shader instance with data binding and rule
+      const waterTankDataBinding = new DataBinding();
+      tagDataBinding
+        .withTargetEntityId(entitySummary.entityId)
+        .withTargetComponentName('WaterTank')
+        .withTargetProperty('flowRate1');
+
+      const modelShader: ModelShader = new ModelShader()
+        .withValueDataBinding(waterTankDataBinding)
+        .withRuleId(waterTankFlowRuleName);
+
+      waterTankNode.addModelShader(modelShader);
+      equipmentNode.addChildNode(waterTankNode);
+    }
+  };
+
+  await factory.updateSceneForEntities(workspaceId, addModelAndRuleForEntity, [listEntitiesFilter]);
+
+  twinMakerScene.addRootNodeIfNameNotExist(rootNode);
+
+  // Save the changes to the Scene
+  await factory.save(twinMakerScene);
```

Comment:
IIUC the behavior here is to not overwrite the scene if it already exists? If so, we may want to update some of the messages in the sample or adjust to early exit if scene already exists so not making any changes. e.g. I ran and got the following and thought the updated scene was saved to cloud but saw no change there:

```
Scene already exists, loading scene...
Creating/Editing Cookie Factory scene...
Adding Cookie Lines...
Adding Mixers and WaterTank...
Saving scene scene01...
Scene saved!
```

Reply:
Looking at `save` and `overrideSave` there wasn't any difference in implementation. I will remove `overrideSave` since `save` will already override the scene if it already exists. Added a comment to clarify this to `save()` in `scene_factory.ts`.

The method `loadOrCreateSceneIfNotExists` will load an existing scene into memory if it already exists. I don't think it makes sense to have a method to save a scene only if it doesn't exist.